Std/Lean/Expr.lean

```diff
@@ -40,6 +40,14 @@ def lambdaArity : Expr → Nat
   | lam _ _ b _ => 1 + lambdaArity b
   | _ => 0
 
+/-- If the expression is a constant, return that name. Otherwise return `Name.anonymous`. -/
+def constName (e : Expr) : Name :=
+  e.constName?.getD Name.anonymous
+
+/-- Return the function (name) and arguments of an application. -/
+def getAppFnArgs (e : Expr) : Name × Array Expr :=
+  withApp e fun e a => (e.constName, a)
+
```

Comment:
Would these and the Std/Lean/Name operations be better as utility functions inside the Simps tactic?  They seem to make assumptions about the names that may hold in the context they are used in that tactic, but could mask errors in other contexts.

Reply:
This one I would prefer to keep out "in public", as it is very thoroughly used all over tactics in Mathlib. (And e.g. I will use it in `omega`.) It's basically the standard interface for decomposing an expression, although probably should be deprecated in favour of Kyle's recent https://github.com/leanprover/lean4/pull/2900.

Reply:
Ok, that makes sense.  This does seem like a common need to get the head function and arguments.  Use of anonymous is appropriate given that need.